packages/aws-cdk-lib/aws-efs/lib/efs-file-system.ts

```diff
@@ -767,7 +769,10 @@ export class FileSystem extends FileSystemBase {
       lifecyclePolicies.push({ transitionToArchive: props.transitionToArchivePolicy });
     }
 
-    const oneZoneAzName = props.vpc.availabilityZones[0];
+    // if props.vpcSubnets.availabilityZones is defined, select the first one as the zone otherwise
+    // the first AZ of the VPC.
+    const oneZoneAzName = props.vpcSubnets?.availabilityZones ?
+      props.vpcSubnets.availabilityZones[0] : props.vpc.availabilityZones[0];
```

Comment:
Maybe add a check for

```ts
props.vpc.availabilityZones.includes(props.vpcSubnets.availabilityZones[0])
```

Reply:
OK this is a little tricky because `vpc.availabilityZones` could be dummy values if the context was not generated by the context provider. Let's only check that when we are confident they are neither dummy values nor unresolved tokens.

Reply:
Yeah good call, I only saw the unresolved tokens coming